File: python/3_longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        # Check input
        if not s:
            return 0

        # Initialization
        ans = 0
        table = {}
        j = 0

        for i, string in enumerate(s):
            if string in table:
                j = max(j, table[string] + 1)

            table[string] = i
            ans = max(ans, (i - j + 1))

        return ans

File: go/3_longest-substring-without-repeating-characters.go
func lengthOfLongestSubstring(s string) int {
	if len(s) == 0 {
		return 0
	}

	dict := make(map[rune]int)
	var start float64 = 0
	var ans float64 = 0

	for i, c := range s {
		if dict[c] != 0 {
			start = math.Max(start, float64(dict[c]))
		}

		dict[c] = i + 1
		ans = math.Max(ans, float64(i-int(start)+1))
	}

	return int(ans)
}